go/vt/vtgate/planbuilder/testdata/union_cases.txt

```diff
@@ -1435,6 +1470,10 @@ Gen4 error: nesting of unions at the right-hand side is not yet supported
   "Original": "(select id from user union select id from music order by id) union select 1 from unsharded",
   "Instructions": {
     "OperatorType": "Distinct",
+    "Collations": [
+      "1: binary"
+    ],
+    "ResultColumns": 1,
```

Comment:
The looks like the planner knows about collation, then the query should not contain weight_string column